File: go-sdk/pkg/models/operations/getcollectionsiditems.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetCollectionsIDItemsPathParams struct {
	CollectionID string `pathParam:"style=simple,explode=false,name=collection_id"`
}

type GetCollectionsIDItemsQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
	Limit  *int64   `queryParam:"style=form,explode=true,name=limit"`
	Offset *int64   `queryParam:"style=form,explode=true,name=offset"`
}

type GetCollectionsIDItemsRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetCollectionsIDItemsPathParams
	QueryParams GetCollectionsIDItemsQueryParams
}

type GetCollectionsIDItemsResponse struct {
	ClientError *shared.ClientError
	ContentType string
	Items       *shared.Items
	StatusCode  int64
}

File: sdks/ts/src/sdk/models/operations/getcollectionsiditems.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCollectionsIdItemsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=collection_id" })
  collectionId: string;
}


export class GetCollectionsIdItemsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;
}


export class GetCollectionsIdItemsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetCollectionsIdItemsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCollectionsIdItemsQueryParams;
}


export class GetCollectionsIdItemsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  items?: shared.Items;

  @SpeakeasyMetadata()
  statusCode: number;
}
